operator/v1/types_console.go

```diff
@@ -298,6 +301,28 @@ type Perspective struct {
 	// visibility defines the state of perspective along with access review checks if needed for that perspective.
 	// +kubebuilder:validation:Required
 	Visibility PerspectiveVisibility `json:"visibility"`
+	// pinnedResources defines the list of default pinned resources that users will see on the perspective navigation if they have not customized these pinned resources themselves.
+	// The list of available Kubernetes resources could be read via `kubectl api-resources`.
+	// The console will also provide a configuration UI and a YAML snippet that will list the available resources that can be pinned to the navigation.
+	// Incorrect or unknown resources will be ignored.
+	// +optional
+	PinnedResources []PinnedResourceReference `json:"pinnedResources"`
+}
+
+// PinnedResourceReference includes the group, version and type of resource
+type PinnedResourceReference struct {
+	// Group is the API Group of the Resource.
+	// Example: "apps", "build.openshift.io", etc.
+	// +optional
+	Group string `json:"group"`
+	// Version is the API Version of the Resource.
+	// Example: "v1", "v1beta1", etc.
+	// +kubebuilder:validation:Required
+	Version string `json:"version"`
+	// Resource is the type that is being referenced
+	// Example: "Deployment", "DeploymentConfig", "Pod", etc.
```

Comment:
So, should I change 'Resource' to 'Kind' ?

Reply:
The Group , Version and Kind of the Resource will be used in our frontend code to identify if the Resource exists and then display it on the Navigation

Reply:
Can you link me to where you're doing the mapping from Kind to the resource? A code example would help my understanding

Normally, in Go, you would use a restmapper, which fetches information from the API server about available API resources, to map the resource to a kind, your suggestion suggests you're doing this the other way around which is not necessarily guaranteed to produce the correct result

Reply:
This is how we are passing the the default Pins in the extension plugin
```
"defaultPins": [
        {
          "group": "",
          "version": "v1",
          "kind": "ConfigMap"
        },
        {
          "group": "",
          "version": "v1",
          "kind": "Secret"
        }
      ],
```

and this the logic:

```
export const referenceForExtensionModel = (model: ExtensionK8sGroupModel): GroupVersionKind =>
  referenceForGroupVersionKind(model?.group || 'core')(model?.version)(model?.kind);
 
const defaultPins = useMemo(
    () =>
      perspectiveExtensions.reduce(
        (acc, e) => ({
          ...acc,
          [e.properties.id]: (e.properties.defaultPins || []).map((gvk) =>
            referenceForExtensionModel(gvk),
          ),
        }),
        {},
      ),
    [perspectiveExtensions],
  );
```

and this is how it is saved in the user settings config map
```
"dev":["core~v1~ConfigMap","core~v1~Secret"]}
```

Reply:
@JoelSpeed Your lowercase "deployments" confuses me as well. You said a GVR refers exactly and directly a resource. Isn't "deployments" one of the aliases?

Reply:
> Your lowercase "deployments" confuses me as well. You said a GVR refers exactly and directly a resource. Isn't "deployments" one of the aliases?

No, `deployments` is the correct and canonical resource name, `Deployment` is an alias to `deployments`. 

For example, if you look at the `kubectl api-resources` for `deployments`, note the Name field (the unique identifier) is `deployments` and the `Kind` field, is later, because it's an alias
```
NAME                              SHORTNAMES   APIVERSION                             NAMESPACED   KIND
deployments                       deploy       apps/v1                                true         Deployment
```

Somewhere in the code you are doing, you are converting `Deployment` to `deployments`, except you're not guaranteed that that's correct. You need this conversion to form a valid API request, so you may not be explicitly conscious of it, this is why I asked where the code was, so that I can have a look and point out where that conversion is happening

Reply:
In https://github.com/openshift/console/blob/3ea52056b9f8e486599353fc044336eae4c2a6c0/frontend/public/module/k8s/get-resources.ts#L139-L200 you are performing API discovery to look up the CRDs and API types that exist within the cluster.

You're creating a model map here which maps the name of the resources (ie what we reference to as `resource` here) as the `plural` of the model. Note how this logic is all based on the plural version being the name of the resource, as that's the unique identifier.

When you lookup the models from your cache, you should look them up by the GVR as this is guaranteed to be unique, looking at the current implementation, you are looking them up by GVK, but that's not guaranteed to be unique